tests/integration/test.conflicts.js

```diff
@@ -116,6 +116,37 @@ adapters.forEach(function (adapter) {
       });
     });
 
+
+    it('force put ok', function () {
+      var db = new PouchDB(dbs.name);
+      var docId = "docId";
+      var rev1, rev2, rev3, rev2_;
+      // given
+      return db.put({_id: docId, update:1}).then(function (result) {
+        rev1 = result.rev;
+        return db.put({_id: docId, update:2.1, _rev: rev1});
+      }).then(function (result) {
+        rev2 = result.rev;
+        return db.put({_id: docId, update:3, _rev:rev2});
+      })
+      // when
+      .then(function (result) {
+        rev3 = result.rev;
+        return db.put({_id: docId, update:2.2, _rev: rev1}, {force: true});
+      })
+      // then
+      .then(function (result) {
+        rev2_ = result.rev;
+        rev2_.should.not.equal(rev3);
+        rev2_.substring(0, 2).should.equal('2-');
+        should.exist(result.ok, 'update based on nonleaf revision');
+        return db.get(docId, {conflicts: true});
+      }).then(function (doc) {
+        doc._rev.should.equal(rev3);
+        doc._conflicts.should.eql([rev2_]);
```

Comment:
1, How can I get the whole rev tree? I can't find the public api to achieve that.

2, I assume the `_rev` user provide when doing force put is valid. Conceptually the only thing need to do is create a new child node for the `_rev` node, so it doesn't matter the new child node is on the 2nd level or 3rd level. I will add a new test case to cover that.